File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.cpp
class Solution {
public:
    int longestMonotonicSubarray(vector<int>& nums) {
        int n=nums.size();
        int ans=1;
        int cnt=1;
        for(int i=1;i<n;i++){
            if(nums[i]>nums[i-1]){
                cnt++;
            }
            else {
            ans=max(ans,cnt);
            cnt=1;
            }
        }
        ans=max(ans,cnt);
        cnt=1;
        for(int i=n-2;i>=0;i--){
            if(nums[i]>nums[i+1]){
                cnt++;
            }
            else {
            ans=max(ans,cnt);
            cnt=1;
            }
        }
        ans=max(ans,cnt);
        return ans;
    }
};

File: 3372-longest-strictly-increasing-or-strictly-decreasing-subarray/longest-strictly-increasing-or-strictly-decreasing-subarray.py
class Solution(object):
    def longestMonotonicSubarray(self, nums):
        ans=1
        n=len(nums)
        cnt=1
        cnt1=1
        for i in range(1,n):
            if nums[i]>nums[i-1]:
                cnt+=1
            else:
                ans=max(cnt,ans)
                cnt=1
                
            if nums[i]<nums[i-1]:
                cnt1+=1
            else:
                ans=max(cnt1,ans)
                cnt1=1
            
        ans=max(ans,max(cnt1,cnt))
        return ans
        

        